giturlparse/parser.py

```diff
@@ -69,36 +69,30 @@ def parse(self):
             'name': None,
             'owner': None,
         }
-
-        regexp = (r'^(https?|git|ssh|rsync)\://'
-                  '(?:(.+)@)*'
-                  '([a-z0-9_.-]*)'
-                  '[:/]*'
-                  '([\d]+){0,1}'
-                  '(/(.+)/(.+).git)')
-        m1 = re.search(regexp, self._url)
-
-        regexp = (r'^(?:(.+)@)*'
-                  '([a-z0-9_.-]*)[:/]*'
-                  '([\d]+){0,1}'
-                  '([:/](.+)/(.+).git)')
-        m2 = re.search(regexp, self._url)
-
-        if m1:
-            d['pathname'] = m1.group(5)
-            d['protocol'] = m1.group(1)
-            d['resource'] = m1.group(3)
-            d['user'] = m1.group(2)
-            d['port'] = m1.group(4)
-            d['name'] = m1.group(7)
-            d['owner'] = m1.group(6)
-        elif m2:
-            d['pathname'] = re.sub(r'^:', '', m2.group(4))
-            d['protocol'] = 'ssh'
-            d['resource'] = m2.group(2)
-            d['user'] = m2.group(1)
-            d['name'] = m2.group(6)
-            d['owner'] = m2.group(5)
+        regexes = [
+            (r'^(?P<protocol>https?|git|ssh|rsync)\://'
+                '(?:(?P<user>.+)@)*'
+                '(?P<resource>[a-z0-9_.-]*)'
+                '[:/]*'
+                '(?P<port>[\d]+){0,1}'
+                '(?P<path>\/(?P<owner>.+)/(?P<repo>.+).git)'),
+            (r'^(?:(?P<user>.+)@)*'
+                '(?P<resource>[a-z0-9_.-]*)[:/]*'
+                '(?P<port>[\d]+){0,1}'
+                '[:](?P<path>\/?(?P<owner>.+)/(?P<repo>.+).git)')
+                ]
+        for regex in regexes:
+            if re.search(regex, self._url):
+                m = re.search(regex, self._url)
+                gdict = m.groupdict()
```

Comment:
```python
d['user'] = gdict.get('user', None)
```

you are playing with two dicts with the same member names.
That is a dead giveaway that you should be doing a loop, not playing with each of the members by name.

but more importantly, you can do bulk `dict` manipulations.

```python
d.update( m.groupdict() )
```

Then you need to solve the `protocol` = `ssh` fallback a smarter way.